### ISSUE TITLE
Feature request: multiple databases simultaneously or easy switch 

### ISSUE THREAD

**Issue by Edwin-VC** [Kunzisoft/KeePassDX#209]
I have a DB for personal use and one for work. I would like to be able to have them both open at the same time or at least be able to easily switch between the database. The desktop version I use has 2 (or more) tabs.

**Comment by J-Jamet** [comment]
It will be difficult to open 2 databases at the same time, but the navigation and opening system is being rebuilt, so it will be easier to switch between the databases.

**Comment by Adrianzo** [comment]
Maybe inside a database there could be a button that closes the current opened database and takes you back to the main screen.  That alone would help a lot for those of us who have more than one databse on our phones. Thanks in advance!

**Comment by J-Jamet** [comment]
@Adrianzo You mean like a closed padlock button always visible in the actionbar? Haven't you seen him? : D

**Comment by trymeouteh** [comment]
This will be awesome, just like how KeePassXC can open multiple databases at once. I think this will be useful for those who use KeePassDX as a password manager and TOTP manager and use two databases, one for passwords and another one for TOTP.

**Comment by J-Jamet** [comment]
I'm thinking about the architecture of the app to open several databases at the same time, the quick unlock may be a complementary solution. Above all, the functionalities have to be harmonized correctly in the code, which is not at all easy.

**Comment by adatum** [comment]
Yes! I didn't realize this feature was missing and was confused for a while. Luckily, Keepass2Android supports opening multiple databases. I have both password managers installed, so that's one workaround. It would be great for KeePassDX to gain this feature too.

**Comment by wjuergen** [comment]
would be a great feature for the pro

**Comment by seniorm0ment** [comment]
Came here to request this feature, seems I'm not the only one interested. This is a major setback imo.
Also hoping it comes with support for AutoOpen to work with it.
https://keepassxc.org/docs/KeePassXC_UserGuide.html#_automatic_database_opening

**Comment by seniorm0ment** [comment]
Any updates on this @J-Jamet? See you referenced an issue from Aug and said you were refactoring the codebase.
Do you have an *estimated* month you might get this up and released?

Very very tedious to open a db, then need to get in the other one so you have to close it, and then afterwards get something from the original db so reopen that.

**Comment by J-Jamet** [comment]
No time estimate, I work when I can, I refactor as I go and I know the goal so be patient.

**Comment by Ferk** [comment]
> @Adrianzo You mean like a closed padlock button always visible in the actionbar? Haven't you seen him? : D
> 
> Edit: The close button is now independent at the bottom left for better visibility.

Just to update on the current state:  when the App is searching for a matching entry (when configured as the default Android password manager) locking the database closes the Intent and goes back to the browser (or whichever other App requested for password).

It's unclear what's the current user flow to switch database from a password prompt in a separate App (such as a browser).

What I currently do is manually open the KeyPassDX App and unlock the database I want, then switch back to the app that needs the password.

**Comment by Zhen-LinHuo** [comment]
Can open and search in multiple databases also help a lot in multi-device sync: a possible solution is each device will apply its local change to the database belongs to it, and also read records from other database (which "belongs" to other device), so that they can update records without overwrite other databases
